vendor/runner/runner.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const glob = require('glob');
+const uniq = require('lodash.uniq');
+
+/**
+ * Load a collection of ServiceTester objects and register them with Mocha.
+ */
+class Runner {
+  /**
+   * Function to invoke before each test. This is a stub which can be
+   * overridden on instances.
+   */
+  beforeEach () {}
+
+  /**
+   * Prepare the runner by loading up all the ServiceTester objects.
+   */
+  prepare () {
+    this.testers = glob.sync(`${__dirname}/../*.js`).map(require);
+    this.testers.forEach(tester => {
+      tester.beforeEach = () => { this.beforeEach(); };
+    });
+  }
+
+  _testersWithVendor (vendor) {
+    return this.testers.filter(t => t.id.toLowerCase() === vendor);
+  }
+
+  /**
+   * Limit the test run to the specified vendors.
+   *
+   * @param vendors An array of vendor names to run
+   */
+  only (vendors) {
+    const normalizedVendors = uniq(vendors.map(v => v.toLowerCase()));
```

Comment:
Yeah, it's annoying that sets don't have useful utility methods. Not only `.difference` and `.union`, but even basic methods like `.filter` are missing. :(

At Facebook we have a `differenceSets` method which looks like:
```js
/**
 * Returns a set containing all of the values from the first set that are not
 * present in any of the subsequent sets.
 */
function differenceSets<TValue>(
  set: Set<TValue>,
  ...setsWithValuesToRemove: Array<Set<TValue>>
): Set<TValue> {
  return filterSet(
    set,
    value => !setsWithValuesToRemove.some(
      otherSet => otherSet.has(value),
    ),
  );
}
```

Which uses `filterSet`:
```js
/**
 * Returns a set containing all of the values from the original set where
 * the given callback returned true.
 */
function filterSet<TValue>(
  set: Set<TValue>,
  callback: (value: TValue) => boolean,
): Set<TValue> {
  const result = new Set();
  for (const value of set) {
    if (callback(value)) {
      result.add(value);
    }
  }

  return result;
}
```

They're part of the internal version of fbjs (https://github.com/facebook/fbjs/tree/master/packages/fbjs/src/functional) but it looks like they haven't been synced out to GitHub yet